pkg/restore/restore.go

```diff
@@ -1344,6 +1381,17 @@ func (ctx *restoreContext) restoreItem(obj *unstructured.Unstructured, groupReso
 		return warnings, errs
 	}
 
+	// if it should restore status, run a UpdateStatus
+	if shouldRestoreStatus {
+		obj.SetResourceVersion(createdObj.GetResourceVersion())
```

Comment:
Comment 5: set the status value of obj from the objWithStatus or the kept the status value